File: mdc.java
import java.util.Scanner;

public class mdc {
    public static int calcularMDC(int a, int b) {
        if (b == 0) {
            return a;
        } else {
            return calcularMDC(b, a % b);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite o primeiro número: ");
        int a = scanner.nextInt();

        System.out.print("Digite o segundo número: ");
        int b = scanner.nextInt();

        int mdc = calcularMDC(a, b);
        System.out.println("O máximo divisor comum entre " + a + " e " + b + " é: " + mdc);

        scanner.close();
    }
}

File: mdc.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function mdc(a, b) {
  if (b === 0) {
    return a;
  } else {
    return mdc(b, a % b);
  }
}

rl.question("Digite o primeiro número: ", (numero1) => {
  rl.question("Digite o segundo número: ", (numero2) => {
    let md = mdc(numero1, numero2);
    console.log("O mdc de " + numero1 + " e " + numero2 + " é " + md + ".");
    rl.close();
  });
});
